app/controllers/users_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen-string-literal: true
+
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    @user.save!
+    redirect_to @user, status: :created
+  rescue ActiveRecord::RecordInvalid
+    render :new, status: :unprocessable_entity
+  end
+
+  def show
+    @user = User.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    flash[:alert] = 'User Not Found'
+    render status: :not_found
+  end
```

Comment:
I really don't think that a service object is necessary - there is no logic besides a simple model method call (eg: `User#find`, `User#new`).

I was worried with using `rescue_from`because of the implicit template calling after the method is done. Gonna fix it